components/brave_wayback_machine/brave_wayback_machine_tab_helper.cc

```diff
@@ -58,10 +57,10 @@ void BraveWaybackMachineTabHelper::DidFinishNavigation(
       return;
 
     // Create infobar in the next loop for not blocking navigation.
-    PostTask(FROM_HERE,
-             { base::CurrentThread(), base::TaskPriority::BEST_EFFORT },
-             base::BindOnce(&BraveWaybackMachineTabHelper::CreateInfoBar,
-                            weak_factory_.GetWeakPtr()));
+    DCHECK(base::SequencedTaskRunnerHandle::IsSet());
+    base::SequencedTaskRunnerHandle::Get()->PostTask(
```

Comment:
basically it always happens on the UI thread, so DCHECK is sort of not needed. And you can just post to UI.